src/switch_rtp.c

```diff
@@ -2199,6 +2279,11 @@ static int check_rtcp_and_ice(switch_rtp_t *rtp_session)
 				rtcp_generate_report_block(rtp_session, rtcp_report_block, nack_dup);
 				rtp_session->rtcp_send_msg.header.count = 1; /* reception report block count */
 				stats->sent_pkt_count = 0;
+				/* send event for audio, or video if flag set */
+				if (rtp_session->flags[SWITCH_RTP_FLAG_VIDEO_FIRE_SEND_RTCP_EVENT] || 
```

Comment:
we had an internal discussion about this patch and we need more clarifications about the run conditions here. I thought I made it clear in my previous comments that we don't want this code run by default.  Also the naming of the vars/flags is confusing, eg: "SWITCH_RTP_FLAG_VIDEO_FIRE_SEND_RTCP_EVENT" - is this meant to work only for video ? according to this conditions it fires for audio every time.

Reply:
just make it so that it fires for both audio and video with the same code - and use the flag to disable all the patch.

Reply:
@dragos-oancea I don't fully understand this.  `RECV_RTCP_MESSAGE` fires by default now, so I don't understand why `SEND_RTCP_MESSAGE` wouldn't?  And yes, the flag is to enable the `SEND_RTCP_MESSAGE` for video, in addition to audio.

Reply:
My interpretation is that video is the greater concern, with respect to load on the events system.  If audio is also a concern, I propose that we have a config setting, and a channel var to override, that sets the rtp session flag, that enables RTCP events either direction, for each audio or video separately.  so, two flags:
SWITCH_RTP_FLAG_RTCP_EVENTS_VIDEO
SWITCH_RTP_FLAG_RTCP_EVENTS_AUDIO
The flag toggles send/recv events together, for video and audio respectively.

With a config and/or channel var, plus rtcp-audio/video-interval-msec, this gives the user pretty fine control over these events, and hence the load on the events system.  Enable audio, disable video by default.

Reply:
"I don't fully understand this. RECV_RTCP_MESSAGE fires by default now" - this will change soon with upcoming code which already exists but it's not merged yet. The creation and enqueuing of these events will be conditioned by a bool chan var named "fire_rtcp_events". I agree with the two flags, the whole idea is to NOT have this code run by default, even if you rightfully think it's consistent with the existing code.

Reply:
ok, that makes more sense now.  Can you point to that existing code that's not yet merged?  I'll try to make the code here congruent with "fire_rtcp_events".  Please advise.

Reply:
no, I can't point you to private code. all you need to do is enable your whole patch based on a chan var named "fire_rtcp_events".

Reply:
that implies one rtp session flag, SWITCH_RTP_FLAG_FIRE_RTCP_EVENTS, that toggles events for both video and audio, for both send & recv.  agree?

Reply:
@dragos-oancea @rjm2718 it would still be helpful to have a config setting for this as well, as we don't want to *have* to update our applications to send new channel vars, if we're going to do it on every session anyways.

Reply:
@rjm2718  something like this is acceptable, for more fine-tuning:
```
if (switch_channel_var_true(session->channel, "fire_rtcp_events")) {
                       flags[SWITCH_RTP_FLAG_AUDIO_FIRE_SEND_RTCP_EVENT] = 1;
			if (switch_channel_var_true(session->channel, "enable_send_rtcp_message_event_video")) { 
				flags[SWITCH_RTP_FLAG_VIDEO_FIRE_SEND_RTCP_EVENT] = 1;
			}
}
```

Reply:
@noahmehl , config setting ? you can just use the chan vars as "config settings". you can set them in vars.xml and have them on all channels.

Reply:
@dragos-oancea understood.  Thanks!